### PR TITLE
Java worker kitchen sink

### PR DESCRIPTION
**Pull request by Quinn-With-Two-Ns**
Add Java omes worker and kitchen sink workflow

### CODE REVIEW

File: workers/java/io/temporal/omes/Main.java [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+        throw new RuntimeException("Error loading certs", e);
+      }
+
+    } else if (StringUtils.isNotEmpty(clientKeyPath) && StringUtils.isEmpty(clientCertPath)) {
+      throw new RuntimeException("Client cert path must be specified since key path is");
+    } else if (isTlsEnabled) {
+      try {
+        sslContext = SimpleSslContextBuilder.noKeyOrCertChain().build();
+      } catch (SSLException e) {
+        // TODO Auto-generated catch block
```
Comment (Sushisource) [inline workers/java/io/temporal/omes/Main.java:145]: ?

File: loadgen/kitchensink/kitchen_sink.pb.go [hunk 2]
```diff
@@ -3393,11 +3393,12 @@ var file_kitchen_sink_proto_rawDesc = []byte{
...
-	0x2e, 0x67, 0x69, 0x74, 0x68, 0x75, 0x62, 0x2e, 0x63, 0x6f, 0x6d, 0x2f, 0x74, 0x65, 0x6d, 0x70,
-	0x6f, 0x72, 0x61, 0x6c, 0x69, 0x6f, 0x2f, 0x6f, 0x6d, 0x65, 0x73, 0x2f, 0x6c, 0x6f, 0x61, 0x64,
-	0x67, 0x65, 0x6e, 0x2f, 0x6b, 0x69, 0x74, 0x63, 0x68, 0x65, 0x6e, 0x73, 0x69, 0x6e, 0x6b, 0x62,
-	0x06, 0x70, 0x72, 0x6f, 0x74, 0x6f, 0x33,
+	0x12, 0x0b, 0x0a, 0x07, 0x41, 0x42, 0x41, 0x4e, 0x44, 0x4f, 0x4e, 0x10, 0x02, 0x42, 0x42, 0x0a,
+	0x10, 0x69, 0x6f, 0x2e, 0x74, 0x65, 0x6d, 0x70, 0x6f, 0x72, 0x61, 0x6c, 0x2e, 0x6f, 0x6d, 0x65,
+	0x73, 0x5a, 0x2e, 0x67, 0x69, 0x74, 0x68, 0x75, 0x62, 0x2e, 0x63, 0x6f, 0x6d, 0x2f, 0x74, 0x65,
+	0x6d, 0x70, 0x6f, 0x72, 0x61, 0x6c, 0x69, 0x6f, 0x2f, 0x6f, 0x6d, 0x65, 0x73, 0x2f, 0x6c, 0x6f,
+	0x61, 0x64, 0x67, 0x65, 0x6e, 0x2f, 0x6b, 0x69, 0x74, 0x63, 0x68, 0x65, 0x6e, 0x73, 0x69, 0x6e,
+	0x6b, 0x62, 0x06, 0x70, 0x72, 0x6f, 0x74, 0x6f, 0x33,
```
Comment (dandavison) [inline loadgen/kitchensink/kitchen_sink.pb.go:3401]: Just curious what this change is -- why does adding a Java worker cause a modification to the KS proto generated Go code?

File: workers/java/io/temporal/omes/KitchenSinkWorkflowImpl.java [hunk 3]
```diff
@@ -0,0 +1,457 @@
...
+
+public class KitchenSinkWorkflowImpl implements KitchenSinkWorkflow {
+  public static final Logger log = Workflow.getLogger(KitchenSinkWorkflowImpl.class);
+  Map<String, String> state = new HashMap<>();
+  WorkflowQueue<KitchenSink.ActionSet> signalActionQueue = Workflow.newWorkflowQueue(100);
+
+  @Override
+  public Payload execute(KitchenSink.WorkflowInput input) {
+    // Run all initial input actions
+    if (input != null && input.getInitialActionsList().size() > 0) {
```
Comment (dandavison) [inline workers/java/io/temporal/omes/KitchenSinkWorkflowImpl.java:33]: nit: no need to check size, just let the for loop run over empty container?

File: workers/java/io/temporal/omes/KitchenSinkWorkflowImpl.java [hunk 4]
```diff
@@ -0,0 +1,457 @@
...
+          action.getUpsertSearchAttributes();
+      Workflow.upsertSearchAttributes(upsertSearchAttributes.getSearchAttributes());
+    } else if (action.hasSetPatchMarker()) {
+      KitchenSink.SetPatchMarkerAction patchMarker = action.getSetPatchMarker();
+      if (Workflow.getVersion(patchMarker.getPatchId(), -1, 1) == 1) {
+        handleAction(patchMarker.getInnerAction());
+      }
+    } else if (action.hasUpsertMemo()) {
+      // TODO(https://github.com/temporalio/sdk-java/issues/623) Java does not support upsert memo.
+      log.warn("Java does not support upsert memo");
```
Comment (dandavison) [inline workers/java/io/temporal/omes/KitchenSinkWorkflowImpl.java:202]: I wonder whether this warning is going to be more annoying than helpful? We're intentionally submitting upsert-memo actions to all langs, so the statement that sdk-java doesn't support them isn't really a warning that something is wrong with the omes run.

File: workers/java/io/temporal/omes/KitchenSinkWorkflowImpl.java [hunk 5]
```diff
@@ -0,0 +1,457 @@
...
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nonnull;
+import org.slf4j.Logger;
+
+public class KitchenSinkWorkflowImpl implements KitchenSinkWorkflow {
+  public static final Logger log = Workflow.getLogger(KitchenSinkWorkflowImpl.class);
+  Map<String, String> state = new HashMap<>();
+  WorkflowQueue<KitchenSink.ActionSet> signalActionQueue = Workflow.newWorkflowQueue(100);
```
Comment (dandavison) [inline workers/java/io/temporal/omes/KitchenSinkWorkflowImpl.java:28]: How was 100 chosen and what are the consequences of this value?

File: workers/java/io/temporal/omes/KitchenSinkWorkflowImpl.java (hunk 4 above)
Comment (Quinn-With-Two-Ns) [reply workers/java/io/temporal/omes/KitchenSinkWorkflowImpl.java:202]: Yeah. have no issue lowering the log level

File: loadgen/kitchensink/kitchen_sink.pb.go (hunk 2 above)
Comment (Quinn-With-Two-Ns) [reply loadgen/kitchensink/kitchen_sink.pb.go:3401]: This is the proto file description and the proto file did change, it had a Java option added, so I assume that is reflected here.

File: workers/java/io/temporal/omes/KitchenSinkWorkflowImpl.java (hunk 5 above)
Comment (Quinn-With-Two-Ns) [reply workers/java/io/temporal/omes/KitchenSinkWorkflowImpl.java]: The only consequence if we hit the queue limit is the signal handler threads will block on inserting

**Review by Sushisource — APPROVED** [review]

**Pull request merged**